web_poet/page_inputs/client.py

```diff
@@ -33,20 +23,19 @@
 
 
 class HttpClient:
-    """A convenient client to easily execute requests.
+    """Async HTTP client to be used in Page Objects.
 
-    By default, it uses the request implementation assigned in the
-    ``web_poet.request_backend_var`` which is a :mod:`contextvars` instance to
-    download the actual requests. However, it can easily be overridable by
-    providing an optional ``request_downloader`` callable.
+    See :ref:`advanced-requests` for the usage information.
 
-    Providing the request implementation by dependency injection would be a good
-    alternative solution when you want to avoid setting up :mod:`contextvars`
-    like ``web_poet.request_backend_var``.
+    HttpClient doesn't make HTTP requests on itself. It uses either the
+    request function assigned to the ``web_poet.request_backend_var``
+    :mod:`contextvar <contextvars>`, or a function passed via
+    ``request_downloader`` argument of the :meth:`~.HttpClient.__init__` method.
 
-    In any case, this doesn't contain any implementation about how to execute
-    any requests fed into it. When setting that up, make sure that the downloader
-    implementation returns a :class:`~.HttpResponse` instance.
+    Either way, this function should be an ``async def`` function which receives
+    :class:`~.HttpRequest` instance, and either returns :class:`~.HttpResponse`
+    instance, or raises a subclass of :class:`~.HttpError`. You can read
+    more in the :ref:`advanced-downloader-impl` documentation.
```

Comment:
X instance → an X instance

```suggestion
    Either way, this function should be an ``async def`` function which 
    receives an  :class:`~.HttpRequest` instance, and either returns a
    :class:`~.HttpResponse` instance, or raises a subclass of 
    :class:`~.HttpError`. You can read more in the 
    :ref:`advanced-downloader-impl` documentation.
```